packages/odyssey-react-mui/src/labs/SideNav.tsx

```diff
@@ -0,0 +1,601 @@
+  /**
+   * The icon element to display at the start of the Nav Item
+   */
+  startIcon?: ReactElement;
+  /**
+   * The label to display inside the status
+   */
+  statusLabel?: string;
+  target?: string;
+} & (
+  | {
+      /**
+       * Determines if the side nav item is a section header
+       */
+      isSectionHeader: true;
+      href?: never;
+      children?: never;
+    }
+  | {
+      isSectionHeader?: false;
+      href: string;
+      /**
+       * An array of side nav items
+       */
+      children?: SideNavItem[];
+    }
+);
+
+export type SideNavFooterItem = {
+  href: string;
+  id: string;
+  label: string;
+};
+
+const SideNavCollapsedContainer = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "isSideNavCollapsed",
+})(
+  ({
+    odysseyDesignTokens,
+    isSideNavCollapsed,
+  }: {
+    odysseyDesignTokens: DesignTokens;
+    isSideNavCollapsed: boolean;
+  }) => ({
+    backgroundColor: odysseyDesignTokens.HueNeutral300,
+    paddingTop: odysseyDesignTokens.Spacing5,
+    cursor: "pointer",
+    width: isSideNavCollapsed ? "auto" : 0,
+    opacity: isSideNavCollapsed ? 1 : 0,
+    visibility: isSideNavCollapsed ? "visible" : "hidden",
+    transitionProperty: "opacity, visibility, width",
+    transitionDuration: odysseyDesignTokens.TransitionDurationMain,
+    transitionTimingFunction: odysseyDesignTokens.TransitionTimingMain,
+  }),
+);
+
+const SideNavContainer = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "isSideNavCollapsed",
+})(
+  ({
+    odysseyDesignTokens,
+    isSideNavCollapsed,
+  }: {
+    odysseyDesignTokens: DesignTokens;
+    isSideNavCollapsed: boolean;
+  }) => ({
+    backgroundColor: odysseyDesignTokens.HueNeutralWhite,
+    flexDirection: "column",
+    display: "flex",
+    opacity: isSideNavCollapsed ? 0 : 1,
+    visibility: isSideNavCollapsed ? "hidden" : "visible",
+    width: isSideNavCollapsed ? "0" : "100%",
+    transitionProperty: "opacity, visibility, width",
+    transitionDuration: odysseyDesignTokens.TransitionDurationMain,
+    transitionTimingFunction: odysseyDesignTokens.TransitionTimingMain,
+  }),
+);
+
+const SideNavHeaderContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})(({ odysseyDesignTokens }: { odysseyDesignTokens: DesignTokens }) => ({
+  display: "flex",
+  justifyContent: "space-between",
+  alignItems: "center",
+  paddingLeft: odysseyDesignTokens.Spacing4,
+  paddingRight: odysseyDesignTokens.Spacing4,
+  paddingTop: odysseyDesignTokens.Spacing3,
+  paddingBottom: odysseyDesignTokens.Spacing3,
+}));
+
+const CollapseIcon = ({ onClick }: { onClick?(): void }): ReactElement => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  return (
+    <Box
+      sx={{
+        "& > button": {
+          height: "32px",
+          width: "32px",
+          color: odysseyDesignTokens.HueNeutral400,
+        },
```

Comment:
We're resizing this icon right? Can we make sure we don't have magic numbers like this floating around?

Reply:
I am using the values from figma design. Do we keep absolute values somewhere in the tokens? @KevinGhadyani-Okta

Reply:
I asked @bryancunningham-okta to see if he could help with this. Looks like the last thing in this review.

Reply:
@ganeshsomasundaram-okta we don't have absolute(px) values anywhere in the design tokens. I'd use the `Spacing6` variable here